examples/push-file

```diff
@@ -3,11 +3,64 @@
 import argparse
 import json
 import logging
+import kobo.rpmlib
+import os
 
 from chexus import Client, BucketItem, TableItem
 
 LOG = logging.getLogger("push-file")
 
+def get_sigkey(rpmfile):
+    """
+    If the rpm is signed, return the sigkey.
+    rpmfile: path to an rpm
+    """
+    try:
+        rpmhead = kobo.rpmlib.get_rpm_header(rpmfile)
+    except:
+        raise RuntimeError("Cannot read RPM header: %s" % rpmfile)
```

Comment:
I suggest not including this kind of try/except, it mainly obscures the real error.

For example, if someone calls the function with an RPM which doesn't exist, here's the exception which would be raised by this function:

```
RuntimeError: Cannot read RPM header: somefile.rpm
```

If you don't have any try/except at all, what you'll get is instead:

```
IOError: [Errno 2] No such file or directory: somefile.rpm
```

The IOError points directly at the root cause, it tells us *why* the operation failed, but the RuntimeError only tells us *what* operation failed.

I believe in Python 3, there's implicit exception chaining, so the original IOError isn't entirely lost by the try/except.  Still I'm not really a fan of this catching a specific exception only to raise a less-specific one, particularly since users have a tendency to only report the last error message coming from a command.

Reply:
Good point.  I've removed that try/except block.